closure/closure-type-checking.js

```diff
@@ -0,0 +1,59 @@
+#!/usr/bin/env node
+/**
+ * Copyright 2016 Google Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+'use strict';
+
+const closureCompiler = require('google-closure-compiler').gulp();
+const gulp = require('gulp');
+const gutil = require('gulp-util');
+const replace = require('gulp-replace');
+
+/* eslint-disable camelcase */
+gulp.task('js-compile', function() {
+  return gulp.src([
+    'closure/typedefs/*.js',
+    'closure/third_party/*.js',
+    'audits/**/*.js',
+    'metrics/performance/first-meaningful-paint.js'
+  ])
+    // TODO: hack to remove `require`s that Closure currently can't resolve.
+    .pipe(replace(
+        'const DevtoolsTimelineModel = require(\'devtools-timeline-model\');',
+        ''))
+
+    .pipe(closureCompiler({
+      compilation_level: 'SIMPLE',
+      process_common_js_modules: true,
+      new_type_inf: true,
+      checks_only: true,
+      language_in: 'ECMASCRIPT6_STRICT',
+      language_out: 'ECMASCRIPT5_STRICT',
```

Comment:
yeah, no ES6-to-ES6 output. It defaults to `ECMASCRIPT3` for output, though, and for whatever reason it doesnt support all features of ES6 when converting to ES3, so this is an easy fix